src/fu-util.c

```diff
@@ -1831,6 +1855,20 @@ fu_util_update (FuUtilPrivate *priv, gchar **values, GError **error)
 		rel = g_ptr_array_index (rels, 0);
 		if (!fu_util_update_device_with_release (priv, dev, rel, error))
 			return FALSE;
+		if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
+			requires_reboot = TRUE;
+	}
+
+	/* at least one of the updates needed a reboot */
+	if (requires_reboot) {
```

Comment:
In thinking about this a little more, you probably need the override option (--no-reboot-check) for this to allow scripting to still work properly.